src/haddock/modules/refinement/openmm/openmm.py

```diff
@@ -716,11 +716,17 @@ def run_openmm(
             qtemp = n * delta_temp
             integrator.setTemperature(qtemp * kelvin)
             simulation.step(int(eq_steps / nvt_sims))
+        # Try to obtain the degree of freedom
+        try:
+            dof = statereporter._dof
+        except AttributeError:
+            # may not exists if in implicit solvent..
+            dof = system.getNumParticles() * 3
         # Makes sure temperature of the system is reached
         self._stabilize_temperature(
             simulation,
-            qtemp,
-            statereporter._dof,
+            self.params["temperature_kelvin"],
```

Comment:
so before the code was always passing the same temperature within the loop???

Reply:
It is still doing so, I just changed the entry point of the temperature.
Not it is the one that was been set by the user,
Before it was the one set by the user divided by 10, multiplied by 10

Reply:
I see now, thanks!